src/claimtrie.cpp

```diff
@@ -589,6 +589,48 @@ bool CClaimTrie::recursiveCheckConsistency(const CClaimTrieNode* node) const
     return calculatedHash == node->hash;
 }
 
+void CClaimTrie::addToClaimIndex(const std::string& name, const CClaimValue& claim)
+{
+    // sanity checks only
+    claimIndexType::const_iterator itClaim = claimIndex.find(claim.claimId);
+    if (itClaim != claimIndex.end() && itClaim->second.claim != claim)
+        LogPrintf("%s: WARNING: ClaimIndex Conflict on claim id %s\n", __func__, claim.claimId.GetHex());
+
+    CClaimIndexElement element = { name, claim };
+    claimIndex[claim.claimId] = element;
+    LogPrintf("%s: ClaimIndex[%s] updated %s\n", __func__, claim.claimId.GetHex(), name);
+}
+
+void CClaimTrie::removeFromClaimIndex(const CClaimValue& claim)
+{
+    claimIndexType::iterator itClaim = claimIndex.find(claim.claimId);
+    if (itClaim != claimIndex.end())
+    {
+        LogPrintf("%s: ClaimIndex[%s] removing %s\n", __func__, claim.claimId.GetHex(), itClaim->second.name);
+        claimIndex.erase(itClaim);
+    }
+}
+
+bool CClaimTrie::getClaimById(const uint160 claimId, std::string& name, CClaimValue& claim) const
+{
+    claimIndexType::const_iterator itClaim = claimIndex.find(claimId);
+    if (itClaim != claimIndex.end())
+    {
+        name = itClaim->second.name;
+        claim = itClaim->second.claim;
+        return true;
+    }
+
```

Comment:
Hmm, the part that comes after this, where it checks the db does not seem necessary.. 

Correct me if I'm wrong but there should never be a case where if it's not found in claimIndex but its found in leveldb.